tensorflow/compiler/xla/service/pattern_matcher.h

```diff
@@ -1977,14 +1977,17 @@ class HloInstructionPattern {
   }
 
   // Returns true and captures the instruction iff it matches the pattern.
-  bool Match(::xla::HloInstruction* inst, MatchOption option) const {
+  bool Match(::xla::HloInstruction* inst, MatchOption option,
+             bool print_explanation = true) const {
     if (impl_.Match(inst, option)) {
       if (option.capture && matched_inst_) {
         *matched_inst_ = inst;
       }
       return true;
     }
-    EXPLAIN << "\nin " << InstToString(inst);
+    if (print_explanation) {
```

Comment:
This arg doesn't have the right name.  Whether or not we print an explanation is actually part of `option`.

This has to do with printing the relevant instruction.  Choose a different name?